### PR TITLE
Review Bandit reported vulnerabilities

### PR DESCRIPTION
**Pull request by JanuszL**
- silence warnings when applicable
- fix usage of python eval

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- silence warnings when applicable
- fix usage of python eval

### Affected modules and functionalities:
- python

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - numba related tests
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by szkarpinski — APPROVED** [review]

File: dali/python/nvidia/dali/auto_aug/core/_augmentation.py [hunk 1]
```diff
@@ -358,9 +358,9 @@ def _get_param(self, magnitude_bin, num_magnitude_bins, param_device):
             return None
         if magnitude_bin is None:
             raise Exception(
-                f"The augmentation `{self.name}` has `mag_range` specified, so when called, "
-                f"it requires `magnitude_bin` parameter to select the magnitude from the "
-                f"`mag_range`.\nError in augmentation: {self}."
+                f"The augmentation `{self.name}` has `mag_range` specified, "  # nosec B608
+                f" so when called, it requires `magnitude_bin` parameter to select "
```
Comment (stiepan) [inline dali/python/nvidia/dali/auto_aug/core/_augmentation.py:362]: nitpick: there's double space between `specified,` and `so when`.

File: dali/python/nvidia/dali/plugin/numba/experimental/__init__.py [hunk 2]
```diff
@@ -141,11 +142,11 @@ def _get_carray_eval_lambda(self, dtype, ndim):
             eval_string += "shape[{}]".format(i)
             eval_string += ", " if i + 1 != ndim else "), "
         eval_string += "dtype=np.{})".format(_to_numpy[dtype])
-        return njit(eval(eval_string))
+        return njit(ast.literal_eval(eval_string))
```
Comment (stiepan) [inline dali/python/nvidia/dali/plugin/numba/experimental/__init__.py:145]: Does that work?

``` The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis.```

Comment (mzient) [reply dali/python/nvidia/dali/plugin/numba/experimental/__init__.py:145]: I'm quite sure it won't work. We're generating _code_ here. The key part is that it's as trusted as the program itself.

Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/numba/experimental/__init__.py]: Fixed

File: dali/python/nvidia/dali/plugin/numba/experimental/__init__.py
[the remark this answers is not in the document]
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/numba/experimental/__init__.py:1]: Reverted

File: dali/python/nvidia/dali/auto_aug/core/_augmentation.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/auto_aug/core/_augmentation.py]: Fixed

**Review by mzient — APPROVED** [review]

**Pull request merged**